Restrict download menu item to relevant screens

1zaman opened the pull request:
All the various menu items were being defined in `BaseFragmentActivity`, and set invisible by default unless needed by some particular child. This commit unbundles the menus to their appropriate activities, and in the process restricts the download progress wheel menu item to it's appropriate scope. `BaseTabActivity` has been deleted because it caused inconvenience for the new inheritance scheme, and because it was an unneeded abstraction layer that didn't provide much benefit. The test suite hierarchy has been updated to reflect these changes. The base classes are also defined as abstract now.

Previously, the offline menu item display handling was left to each subclass implementation of `BaseFragmentActivity`. This repetitive logic has now been moved inside the base class. It also fixes an issue with the state not being updated on `Activity` resumption, by having `NetworkConnectivityReceiver` post sticky events. Also, the unsafe global static flag for indicating offline state is now removed, and references are replaced with queries to the `NetworkUtil#isConnected()` utility method.  The test suites are also updated to reflect this.

https://openedx.atlassian.net/browse/MA-1665

1zaman commented on VideoLocker/src/main/java/org/edx/mobile/view/VideoListActivity.java:
```diff
@@ -251,23 +250,14 @@ public void updateProgress() {
-        } else {
-            AppConstants.videoListDeleteMode = true;
-        }
+        AppConstants.myVideosDeleteMode = true;
```
Note that using global static variables to share state between `Activity` and `Fragment` as is implemented here is an extremely hacky and unadvisable approach, and should be changed at some point. There are also other issues with the logic for retaining the `Fragment` here, not to mention that every method is wrapped in a generic try catch block that I didn't remove from the unmodified methods for fear that they might mask some hard-to-resolve crash.

miankhalid replied:
create a ticket for that maybe.

1zaman commented:
@miankhalid, @bguertin, please review

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/base/BaseFragmentActivity.java:
```diff
@@ -699,31 +597,34 @@ public boolean showErrorMessage(String header, String message) {
-            }
-        }catch(Exception e){
-            logger.error(e);
+        LinearLayout error_layout = (LinearLayout) findViewById(R.id.error_layout);
```
xml style name, should b `errorLayout`.

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/base/BaseFragmentActivity.java:
```diff
@@ -699,31 +597,34 @@ public boolean showErrorMessage(String header, String message) {
+    }
+
+    public boolean hideErrorMessage() {
+        LinearLayout error_layout = (LinearLayout) findViewById(R.id.error_layout);
```
xml style name again.

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitVideoFragment.java:
```diff
@@ -290,7 +279,7 @@ public void onNegativeClicked() {
                     };
                     MediaConsentUtils.consentToMediaPlayback(getActivity(), dialogCallback, environment.getConfig());
                 }else{
```
missing spaces before and after `else`.

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitVideoFragment.java:
```diff
@@ -529,7 +516,6 @@ public void onOffline() {
     public void onOnline() {
-        AppConstants.offline_flag = false;
         if (!isLandscape) {
 
```
extra line :wink:

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/view/FindCoursesActivity.java:
```diff
@@ -12,6 +12,8 @@
 @ContentView(R.layout.activity_find_courses)
 public class FindCoursesActivity extends FindCoursesBaseActivity {
 
```
extra line :wink:

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/view/MyRecentVideosFragment.java:
```diff
@@ -287,7 +281,6 @@ public void run() {
-            AppConstants.offline_flag = true;
             notifyAdapter();
             videoListView.setOnItemClickListener(adapter);
         }catch(Exception e){
```
spaces :wink:

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/view/MyRecentVideosFragment.java:
```diff
@@ -297,7 +290,6 @@ public void onOffline() {
 
     protected void onOnline() {
         try{
```
spaces :wink:

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/view/MyRecentVideosFragment.java:
```diff
@@ -297,7 +290,6 @@ public void onOffline() {
-            AppConstants.offline_flag = false;
             notifyAdapter();
             videoListView.setOnItemClickListener(adapter);
         }catch(Exception e){
```
spaces :wink:

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/view/VideoListActivity.java:
```diff
@@ -437,7 +403,7 @@ protected void onRestart() {
         if(myVideosFlag){
```
spaces needed.

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/view/VideoListActivity.java:
```diff
@@ -469,7 +435,7 @@ public synchronized void saveCurrentPlaybackPosition(int currentPosition) {
 
     public void onBackPressed() {
-        if(AppConstants.offline_flag){
+        if(!NetworkUtil.isConnected(this)){
```
spaces again.

miankhalid commented on VideoLocker/src/main/java/org/edx/mobile/view/adapters/ChapterAdapter.java:
```diff
@@ -90,21 +91,21 @@ public void onFail(Exception ex) {
-                holder.chapterName.setTextColor(getContext().getResources()
+                holder.chapterName.setTextColor(context.getResources()
                         .getColor(R.color.grey_text_mycourse));
             }else{
```
spaces againn.

miankhalid commented:
some nits otherwise LGTM :+1:

bguertin commented:
LGTM. Happy to see code being removed from those monolithic base activities.

bguertin commented:
Whoops, I just realized those nit-picks weren't fixed. I'm not really worried about white-space and variable names but feel free to clean it up.

1zaman commented:
Most of the formatting nitpicks here were intentionally left alone, because I resolved to not make any additional changes when replacing the `AppConstants.offline_flag` usages, as there were so many of them. It would have taken too long to actually format them manually, and if I used auto-formatting then the diff would have been humongous.

1zaman commented:
I just realized that this comment is no longer valid in the commit message in fc5532d3ca7d6147eecbf220d56f13ad4efd2522:

> `BaseTabActivity` has been deleted because it caused inconvenience for the new inheritance scheme, and because it was an unneeded abstraction layer that didn't provide much benefit.

... as I reverted this part because Mubashar told me that the download progress menu item was not to be displayed in the main screen. I have removed it from the pull request summary.
